tests/compile-and-dump/Singletons/StandaloneDeriving.hs

```diff
@@ -0,0 +1,32 @@
+module Singletons.StandaloneDeriving where
+
+import Data.Singletons.Prelude
+import Data.Singletons.Prelude.Show
+import Data.Singletons.TH
+
+$(singletons [d|
+
+  infixl 6 :*:
+  data T a = a :*: a
+  data S = S1 | S2 deriving Eq
```

Comment:
> So partitionDecs would still handle standalone-deriving for non-Eq classes?

Yes.

> Just add a field which is a list of names for which there is a standalone-deriving for Eq. Or do we need the instance contexts, too?

Yes, you would need the contexts, since you could easy do something like `deriving instance Eq a => Eq (Either a Int)`, but the context you'd get from attaching a `deriving Eq` clause to `data Either a b` would be `(Eq a, Eq b)`. But your point about only needing to keep the bits related to standalone-derived `Eq` instances is well taken. I'll do that (and add a Note, since this is somewhat of an unintuitive design choice).